ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1180,6 +1188,21 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 		newInstance.Status.State = rayv1.Ready
 	}
 
+	// Check if the head node is running and ready with proper reason and message
+	if features.Enabled(features.RayClusterStatusConditions) {
+		isRayHeadRunning, reason, message := utils.CheckRayHeadRunningAndReady(ctx, runtimePods)
```

Comment:
I think we can use [PodReady](https://pkg.go.dev/k8s.io/api/core/v1#PodConditionType) directly. We don't need to check whether the Pod is running, and we don't need to generate the reason and the message ourselves.

Reply:
thanks for the suggestion, I tried using PodReady as much as possible, and notice that pod status condition's reason and message are empty all the time, and since [metav1.Condition](https://pkg.go.dev/k8s.io/apimachinery@v0.30.3/pkg/apis/meta/v1#Condition) need to have an non-empty reason 
```
// reason contains a programmatic identifier indicating the reason for the condition's last transition.
	// Producers of specific condition types may define expected values and meanings for this field,
	// and whether the values are considered a guaranteed API.
	// The value should be a CamelCase string.
	// This field may not be empty.
	// +required
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MaxLength=1024
	// +kubebuilder:validation:MinLength=1
	// +kubebuilder:validation:Pattern=`^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$`
	Reason [string](https://pkg.go.dev/builtin#string) `json:"reason" protobuf:"bytes,5,opt,name=reason"`
```

otherwise will see this error during reconcile
```
{"level":"info","ts":"2024-07-23T00:24:52.098Z","logger":"controllers.RayCluster","msg":"Error updating status","RayCluster":{"name":"rayservice-sample-raycluster-45jnx","namespace":"default"},"reconcileID":"2fb18e10-3cd9-41a0-b9e8-4660293fb634","name":"rayservice-sample-raycluster-45jnx","error":"RayCluster.ray.io \"rayservice-sample-raycluster-45jnx\" is invalid: status.conditions[0].reason: Invalid value: \"\": status.conditions[0].reason in body should be at least 1 chars long","RayCluster":{"apiVersion":"ray.io/v1","kind":"RayCluster","namespace":"default","name":"rayservice-sample-raycluster-45jnx"}}
```
thus i add reason `HeadPodRunningAndReady` and `HeadPodNotReady` respectively